cmd/ocsp-updater/main.go

```diff
@@ -107,96 +243,162 @@ func (updater *OCSPUpdater) processResponse(tx *gorp.Transaction, serial string)
 	return nil
 }
 
-// Produce one OCSP response for the given serial, returning err
-// if anything went wrong. This method will open and commit a transaction.
-func (updater *OCSPUpdater) updateOneSerial(serial string) error {
-	innerStart := time.Now()
-	// Each response gets a transaction. In the future we can increase
-	// performance by batching transactions.
-	// The key thing to think through is the cost of rollbacks, and whether
-	// we should rollback if CA/HSM fails to sign the response or only
-	// upon a partial DB insert.
-	tx, err := updater.dbMap.Begin()
-	if err != nil {
-		updater.log.Err(fmt.Sprintf("OCSP %s: Error starting transaction, aborting: %s", serial, err))
-		updater.stats.Inc("OCSP.Updates.Failed", 1, 1.0)
-		tx.Rollback()
-		// Failure to begin transaction is a fatal error.
-		return FatalError(err.Error())
-	}
+func (updater *OCSPUpdater) newCertLoop() {
+	c := time.Tick(updater.newCertWindow)
```

Comment:
Really, the ticking could happen with a for loop around calling this method. You might want to double check what Tick's behavior is when the code running inside the loop takes longer tahn the Tick interval. I think it drops requests which is maybe not what we want (nor is slowly accreting all of the ticks and having to work through them and never catching up but I don't think that happens, either). What we'd like is a "start as soon as you can and if you finish that one out of time, we'll immediately kick off the next one but not put infinitely many in a queue some where to eat up all of your heap"